src/content/blog/astro-4120.mdx

```diff
@@ -0,0 +1,152 @@
+</BlogSummary>
+
+This release includes the following highlights:
+
+- [**Experimental: Server Islands**](#experimental-server-islands)
+- [**Added `first` and `last` URLs to pagination data**](#added-first-and-last-urls-to-pagination-data)
+- [**Added support for the Shiki's `defaultColor` option**](#added-support-for-the-shikis-defaultcolor-option)
+- [**New `inferRemoteSize` function**](#new-inferremotesize-function)
+
+To upgrade an existing project, use the automated `@astrojs/upgrade` CLI tool. Alternatively, upgrade manually by running the upgrade command for your package manager:
+
+```sh
+# Recommended:
+npx @astrojs/upgrade
+
+# Manual:
+npm install astro@latest
+pnpm upgrade astro --latest
+yarn upgrade astro --latest
+```
+
+## Experimental: Server Islands
+
+In 2021 Astro brought the idea of [islands architecture](https://astro.build/blog/introducing-astro/) to the mainstream, allowing you to create "islands" of interactive client-side components while most of your page is generated statically.
+
+With **Server islands** we are extending this same architecture to the server. Server islands make it easy to combine high performance static HTML and dynamic server-generated components.
+
+Within any given web page you might have content that is:
+
+- Completely static and never changes.
+- Dynamically backed by a database that changes infrequently, but more often than you deploy.
+- Personalized content, tailored to individual users.
+
+Today you must choose 1 caching strategy for all of these types of content, and if the page is a logged-in experience that usually means no caching at all. With Server islands you get the best of both worlds.
+
+<BlogContentImage src={islandDiag} alt="A diagram showing the server island population parts of the page from the server." />
+
+Server islands are used for your most dynamic content; personalized content like the user's avatar, their shopping cart, and product reviews. With these components deferred you're free to more aggressively cache the page itself.
+
+This means that users, logged in or not, will see the most critical parts of the page instantly, as they are caching on Edge CDNs. Fallback content will be visible for a short amount of time before the dynamic islands are then loaded.
+
+Each island is loaded independent from the rest; this means a slower island, such as one connected to a legacy backend, won't delay the rest of personalized content from being seen and interacted with.
+
+### Performance
+
+To test the viability of this idea we built [server-islands.com](https://server-islands.com/). This demo is inspired by the [Next.js Partial Prerendering](https://www.partialprerendering.com/) demo, reskinned to suit Astro's personality. **Server Islands** has a lot in common with Partial Prerendering, especially in its goal of allowing increased cacheability of pages.
```

Comment:
```suggestion
To test the viability of this idea we built [server-islands.com](https://server-islands.com/). This demo is inspired by the [Next.js Partial Prerendering](https://www.partialprerendering.com/) demo, reskinned to suit Astro's personality. **Server Islands** has a lot in common with Partial Prerendering (PPR), especially in its goal of allowing increased cacheability of pages.
```
Just mentioning the acronym that you'll be using throughout the rest of the page here, so that it's clear!